src/main/java/com/openlattice/postgres/mapstores/EntitySetMapstore.java

```diff
@@ -49,18 +46,48 @@ public EntitySetMapstore( HikariDataSource hds ) {
         ps.setArray( 9, flags );
         ps.setArray( 10, partitions );
         ps.setInt( 11, value.getPartitionsVersion() );
+        if ( value.getExpiration() == null ) {
+            ps.setNull( 12, Types.NULL );
+            ps.setNull( 13, Types.NULL );
+            ps.setNull( 14, Types.NULL );
+            ps.setNull( 15, Types.NULL );
+        } else {
+            ps.setObject( 12, value.getExpiration().getTimeToExpiration() );
+            ps.setString( 13, value.getExpiration().getExpirationBase().toString() );
+            ps.setString( 14, value.getExpiration().getDeleteType().toString() );
+            if ( value.getExpiration().getStartDateProperty().isPresent() ) {
+                ps.setObject( 15, value.getExpiration().getStartDateProperty().get() );
+            } else {
+                ps.setNull( 15, Types.NULL );
+            }
+        }
 
         // UPDATE
-        ps.setString( 12, value.getName() );
-        ps.setObject( 13, value.getEntityTypeId() );
-        ps.setString( 14, value.getTitle() );
-        ps.setString( 15, value.getDescription() );
-        ps.setArray( 16, contacts );
-        ps.setArray( 17, linkedEntitySets );
-        ps.setObject( 18, value.getOrganizationId() );
-        ps.setArray( 19, flags );
-        ps.setArray( 20, partitions );
-        ps.setInt( 21, value.getPartitionsVersion() );
+        ps.setString( 16, value.getName() );
+        ps.setObject( 17, value.getEntityTypeId() );
+        ps.setString( 18, value.getTitle() );
+        ps.setString( 19, value.getDescription() );
+        ps.setArray( 20, contacts );
+        ps.setArray( 21, linkedEntitySets );
+        ps.setObject( 22, value.getOrganizationId() );
+        ps.setArray( 23, flags );
+        ps.setArray( 24, partitions );
+        ps.setInt( 25, value.getPartitionsVersion() );
+        if ( value.getExpiration() == null ) {
+            ps.setNull( 26, Types.NULL );
+            ps.setNull( 27, Types.NULL );
+            ps.setNull( 28, Types.NULL );
+            ps.setNull( 29, Types.NULL );
+        } else {
+            ps.setObject( 26, value.getExpiration().getTimeToExpiration() );
+            ps.setString( 27, value.getExpiration().getExpirationBase().toString() );
+            ps.setString( 28, value.getExpiration().getDeleteType().toString() );
```

Comment:
Probably not a big deal but I feel like there's some more optimal way of storing enums in Postgres. I'm also unsure of this, just something to think about as Strings are large to store relative to limited number sets

Reply:
oh interesting. I think I've seen other enums stored in this way, but doesn't mean we have to do it that way! Should i store the ordinal instead?